### PR TITLE
Change Prebuilt OD Configuration and Test Case Metrics

### PR DESCRIPTION
**Pull request by marko-polo-cheno**
Fixes KOL-2846

### CODE REVIEW

File: examples/object_detection_2d/object_detection_2d/seed_test_run.py [hunk 1]
```diff
@@ -116,15 +115,13 @@ def setup_evaluator() -> ObjectDetectionEvaluator:
...
             ThresholdConfiguration(
-                threshold_strategy=ThresholdStrategy.FIXED_03,
+                threshold_strategy=0.3,
                 iou_threshold=0.3,
-                with_class_level_metrics=True,
                 min_confidence_score=0.2,
             ),
             ThresholdConfiguration(
-                threshold_strategy=ThresholdStrategy.F1_OPTIMAL,
+                threshold_strategy="F1-Optimal",
```
Comment (y27choi) [inline examples/object_detection_2d/object_detection_2d/seed_test_run.py]: We should avoid using string here.

File: tests/integration/_experimental/object_detection/test_evaluator_multiclass_fixed.py [hunk 2]
```diff
@@ -882,44 +836,50 @@ def assert_test_case_plots_equals_expected(
...
         test_case=TEST_CASE,
         inferences=TEST_DATA,
         configuration=config,
     )
 
-    assert len(eval.evaluator.threshold_cache) == 0  # empty because not f1 optimal config
-    assert len(eval.evaluator.matchings_by_test_case) != 0
-    assert len(eval.evaluator.matchings_by_test_case[config.display_name()]) != 0
-    assert len(eval.evaluator.matchings_by_test_case[config.display_name()][TEST_CASE.name]) == len(TEST_DATA)
+    print(config)
```
Comment (y27choi) [inline tests/integration/_experimental/object_detection/test_evaluator_multiclass_fixed.py]: Are these intentional?

File: tests/integration/_experimental/object_detection/test_evaluator_multiclass_fixed.py [hunk 3]
```diff
@@ -642,20 +599,15 @@
             ignored=False,
             max_confidence_above_t=0.9,
             min_confidence_above_t=0.6,
-            thresholds=[ScoredClassificationLabel("e", 0.5)],
-            inference_labels=["b", "e"],
+            thresholds=[ScoredClassificationLabel("b", 0.5), ScoredClassificationLabel("e", 0.5)],
```
Comment (y27choi) [inline tests/integration/_experimental/object_detection/test_evaluator_multiclass_fixed.py:598]: Why were these not caught before? 

Should `assert test_sample_metrics == EXPECTED_COMPUTE_TEST_SAMPLE_METRICS` line be checking for the thresholds?

Comment (marko-polo-cheno) [reply tests/integration/_experimental/object_detection/test_evaluator_multiclass_fixed.py:598]: There is a rough edge with the prebuilt OD integration tests being missed in the CI - investigating.

File: examples/object_detection_2d/object_detection_2d/seed_test_run.py (hunk 1 above)
Comment (marko-polo-cheno) [reply examples/object_detection_2d/object_detection_2d/seed_test_run.py]: I think `threshold_strategy: Union[Literal["F1-Optimal"], float]` is a good way for users to define a `number` or `"F1-Optimal"` for the strategy. It feels most intuitive that way for me, much better than `enums`

File: tests/integration/_experimental/object_detection/test_evaluator_multiclass_fixed.py (hunk 2 above)
Comment (marko-polo-cheno) [reply tests/integration/_experimental/object_detection/test_evaluator_multiclass_fixed.py]: The prints will be removed (was debugging CI behaviour)

File: examples/object_detection_2d/object_detection_2d/seed_test_run.py (hunk 1 above)
Comment (y27choi) [reply examples/object_detection_2d/object_detection_2d/seed_test_run.py]: I would avoid using a specific string as a parameter because it's something we would have to look it up every time to get the string match exactly. 

And it looks like if a slightly different version of "F1-Optimal" like "f1-optimal" or "f1" is used then it still accepts it but it will not work as expected.

File: tests/integration/_experimental/object_detection/test_evaluator_multiclass_f1_optimal.py [hunk 4]
```diff
@@ -33,31 +35,29 @@
 from tests.integration._experimental.object_detection.test_evaluator_multiclass_fixed import TEST_DATA
 from tests.integration.helper import fake_locator
 
+kolena.initialize(os.environ["KOLENA_TOKEN"], verbose=True)
```
Comment (ly-kolena) [inline tests/integration/_experimental/object_detection/test_evaluator_multiclass_f1_optimal.py]: Why is this needed? Supposedly the `with_init` fixture, which has `autouse=True`, would have configured this.

File: .circleci/continue_config.yml [hunk 5]
```diff
@@ -194,21 +194,47 @@ jobs:
...
+                    --junitxml=test-results/result.xml $TESTFILES
+      - when:
+          condition:
+            and:
+              - equal:
+                  - "metrics"
+                  - << parameters.extras >>
+              - equal:
+                  - _experimental
+                  - << parameters.pytest-group >>
```
Comment (ly-kolena) [inline .circleci/continue_config.yml]: Nit: do we need this condition, given that we are also using `-m 'metrics'` marker selection?

File: .circleci/continue_config.yml [hunk 6]
```diff
@@ -194,21 +194,47 @@ jobs:
...
+          steps:
+            - run:
+                name: Run experimental metrics integration tests
+                command: |
+                  export KOLENA_TOKEN=${KOLENA_TOKEN}
+                  export KOLENA_CLIENT_BASE_URL=${KOLENA_CLIENT_BASE_URL}
+                  poetry run pytest -m 'metrics' -vv --cov=kolena --cov-branch tests/integration/_experimental
```
Comment (ly-kolena) [inline .circleci/continue_config.yml]: Nit: is `_experimental` optional?

File: tests/integration/_experimental/search/test_embeddings.py [hunk 7]
```diff
@@ -34,6 +34,11 @@
 )
 
 
+def test__dummy_test() -> None:
```
Comment (ly-kolena) [inline tests/integration/_experimental/search/test_embeddings.py]: clean up?

File: tests/unit/_experimental/object_detection/test_configurations.py [hunk 8]
```diff
@@ -87,14 +84,14 @@ def test__object_detection__multiple_configurations__single_class() -> None:
     )
 
     config_one = ThresholdConfiguration(
-        threshold_strategy=ThresholdStrategy.FIXED_05,
+        threshold_strategy=0.5,
         iou_threshold=0.5,
         with_class_level_metrics=False,
```
Comment (ly-kolena) [inline tests/unit/_experimental/object_detection/test_configurations.py]: Should this also be removed?

**Review by ly-kolena — APPROVED** [review]

File: .circleci/continue_config.yml (hunk 5 above)
Comment (marko-polo-cheno) [reply .circleci/continue_config.yml]: I believe so. Correct me if I am wrong, but we need `<< parameters.extras >> == "metrics"` so that the restored cache is suitable

File: tests/unit/_experimental/object_detection/test_configurations.py (hunk 8 above)
Comment (marko-polo-cheno) [reply tests/unit/_experimental/object_detection/test_configurations.py]: Good eye!
